src/App.tsx

```diff
@@ -1,41 +1,67 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import './App.scss';
 import './styles/general.scss';
+
+import { getUserPosts, getPostDetails } from './api/posts';
+import { Post } from './types/Post';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { UserSelect } from './components/UserSelect';
+
+const App: React.FC = () => {
+  const [posts, setPosts] = useState([]);
+  const [postDetails, setPostDetails] = useState({});
+  const [selectedPostId, setSelectedPostId] = useState(0);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+
+  useEffect(() => {
+    getUserPosts(selectedUserId)
+      .then(postsReceived => {
+        setPosts(postsReceived);
+      });
+  }, [selectedUserId]);
+
+  useEffect(() => {
+    if (selectedPostId !== 0) {
+      getPostDetails(selectedPostId)
+        .then(detailsReceived => {
+          setPostDetails(detailsReceived);
+        });
+    }
+  }, [selectedPostId]);
```

Comment:
move this fetching inside PostDetails component (pass just selectedPostId there)